src/binary_basic.ml

```diff
@@ -0,0 +1,471 @@
+open Types
+
+module Input = struct
+  type t =
+    { bytes : bytes
```

Comment:
Using `string` is probably better here as we never mutate it ? In OCaml `string` is simply `bytes` but made immutable...